aleph_alpha_client/image.py

```diff
@@ -93,6 +96,40 @@ def __init__(
         self.cropping = cropping
         self.controls: Sequence[ImageControl] = controls
 
+    @classmethod
+    def from_image_source(
+        cls,
+        image_source: Union[str, Path, bytes],
+        controls: Optional[Sequence[ImageControl]] = None,
+    ):
+        """
+        Abstraction on top of the existing methods of image initialization.
+        If you are not sure what the exact type of your image, but you know it is either a URL, a file path,
+        or a bytes array, just use the method and we will figure out which of the methods of image initialization to use
+        The image will be [center cropped](https://pytorch.org/vision/stable/transforms.html#torchvision.transforms.CenterCrop)
+        """
+        if isinstance(image_source, Path):
+            return cls.from_file(path=str(image_source), controls=controls)
+
+        if isinstance(image_source, str):
+            try:
+                p = urlparse(image_source)
+                if p.scheme:
+                    return cls.from_url(url=image_source, controls=controls)
+            except Exception as e:
+                # we assume that If the string runs into a Exception it isn't not a valid ulr
+                pass
+
+            return cls.from_file(path=image_source, controls=controls)
+
+        elif isinstance(image_source, bytes):
+            return cls.from_bytes(bytes=image_source, controls=controls)
+
+        else:
+            raise TypeError(
+                f"The image source: {image_source} should be either str or bytes"

```

Comment:
or Path ?

Reply:
Added to a comment

Reply:
Hmmm ... shouldn't it also be stated in this error message? Instead of `str or bytes` something like `Path, str or bytes`? Or do you think that this is too much noise?